src/rt/aaA.d

```diff
@@ -152,6 +152,21 @@ private:
         deleted = 0;
         GC.free(obuckets.ptr); // safe to free b/c impossible to reference
     }
+
+    void clear() pure nothrow
+    {
+        foreach (ref b; buckets[firstUsed .. $])
+        {
+            if (b.filled)
+            {
+                b.hash = HASH_DELETED;
+                b.entry = null;
+            }
+        }
+        deleted = used;
+        firstUsed = cast(uint) dim;
+        resize(INIT_NUM_BUCKETS);
```

Comment:
Shrinking the AA to it's initial size defeats one of the main use cases for clear, it should work like appender.clear and keep the buffer so it can be fully refilled w/o resizing.
The AA will adapt it's size to the workload (when there are more deletions) and there is still aa.rehash to shrink w/o deleting.